File: fs/src/test/java/com/radioteria/fs/test/MemoryFileSystemTest.java
package com.radioteria.fs.test;

import com.radioteria.fs.FileSystem;
import com.radioteria.fs.FileSystemException;
import com.radioteria.fs.MemoryFileSystem;
import org.junit.Test;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;

import static junit.framework.TestCase.assertFalse;
import static org.apache.commons.io.IOUtils.copy;
import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class MemoryFileSystemTest {
    private FileSystem fileSystem = new MemoryFileSystem();

    @Test
    public void testFileDoesNotExist() throws FileSystemException {
        assertFalse(fileSystem.has("foo"));
    }

    @Test
    public void testPutGetFile() throws IOException {
        byte[] testData = "Hello, World!".getBytes();
        String contentType = "application/octet-stream";

        fileSystem.put("foo", new ByteArrayInputStream(testData), contentType);

        ByteArrayOutputStream content = new ByteArrayOutputStream();

        copy(fileSystem.get("foo"), content);

        assertArrayEquals(testData, content.toByteArray());
        assertEquals(contentType, fileSystem.getContentType("foo"));
    }

    @Test
    public void testFileExists() throws IOException {
        byte[] testData = "Hello, World!".getBytes();

        fileSystem.put("foo", new ByteArrayInputStream(testData), "application/octet-stream");

        assertTrue(fileSystem.has("foo"));
    }
}

File: src/test/kotlin/com/radioteria/test/fs/MemoryFileSystemTest.kt
package com.radioteria.test.fs

import com.radioteria.fs.MemoryFileSystem
import org.junit.Test

import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream

import junit.framework.TestCase.assertFalse
import org.apache.commons.io.IOUtils.copy
import org.junit.Assert.assertArrayEquals
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue

class MemoryFileSystemTest {
    private val fileSystem = MemoryFileSystem()

    @Test
    fun testFileDoesNotExist() {
        assertFalse(fileSystem.has("foo"))
    }

    @Test
    fun testPutGetFile() {
        val testData = "Hello, World!".toByteArray()
        val contentType = "application/octet-stream"

        fileSystem.put("foo", ByteArrayInputStream(testData), contentType)

        val content = ByteArrayOutputStream()

        copy(fileSystem.get("foo"), content)

        assertArrayEquals(testData, content.toByteArray())
        assertEquals(contentType, fileSystem.getContentType("foo"))
    }

    @Test
    fun testFileExists() {
        val testData = "Hello, World!".toByteArray()

        fileSystem.put("foo", ByteArrayInputStream(testData), "application/octet-stream")

        assertTrue(fileSystem.has("foo"))
    }
}
